yt/frontends/cholla/io.py

```diff
@@ -0,0 +1,54 @@
+from yt.utilities.io_handler import BaseIOHandler
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+
+class ChollaIOHandler(BaseIOHandler):
+    _particle_reader = False
+    _dataset_type = "cholla"
+
+    def _read_particle_coords(self, chunks, ptf):
+        # This needs to *yield* a series of tuples of (ptype, (x, y, z)).
+        # chunks is a list of chunks, and ptf is a dict where the keys are
+        # ptypes and the values are lists of fields.
+        raise NotImplementedError
+
+    def _read_particle_fields(self, chunks, ptf, selector):
+        # This gets called after the arrays have been allocated.  It needs to
+        # yield ((ptype, field), data) where data is the masked results of
+        # reading ptype, field and applying the selector to the data read in.
+        # Selector objects have a .select_points(x,y,z) that returns a mask, so
+        # you need to do your masking here.
+        raise NotImplementedError
+
+    def _read_fluid_selection(self, chunks, selector, fields, size):
+        # This needs to allocate a set of arrays inside a dictionary, where the
+        # keys are the (ftype, fname) tuples and the values are arrays that
+        # have been masked using whatever selector method is appropriate.  The
+        # dict gets returned at the end and it should be flat, with selected
+        # data.  Note that if you're reading grid data, you might need to
+        # special-case a grid selector object.
+        # Also note that "chunks" is a generator for multiple chunks, each of
+        # which contains a list of grids. The returned numpy arrays should be
+        # in 64-bit float and contiguous along the z direction. Therefore, for
+        # a C-like input array with the dimension [x][y][z] or a
+        # Fortran-like input array with the dimension (z,y,x), a matrix
+        # transpose is required (e.g., using np_array.transpose() or
+        # np_array.swapaxes(0,2)).
+
+        # This method is not abstract, and has a default implementation
+        # in the base class.However, the default implementation requires that the method
+        # io_iter be defined
+        data = {}
+        h5f = h5py.File(self.ds.parameter_filename, "r")
+        for field in fields:
+            ftype, fname = field
+            data[field] = h5f[fname][:].flatten().astype("=f8")
+        h5f.close()
+        return data
```

Comment:
This implementation worked for me
```suggestion
    def _read_fluid_selection(self, chunks, selector, fields, size):
        data = {}
        offset = 0

        with h5py.File(self.ds.parameter_filename, "r") as fh:
            for field in fields:
                ftype, fname = field
                data[ftype, fname] = np.empty(size, dtype="float64")
                for chunk in chunks:
                    for grid in chunk.objs:
                        values = fh[fname][:].astype("=f8")
                        offset += grid.select(selector, values, data[field], offset)
        return data
```